analysis/src/github.rs

```diff
@@ -83,11 +120,22 @@ impl GitHubAnalyzer {
         // Get Overall stats for a given repo
         let repo_stats = self.get_github_repo_stats(&repo_fullname)?;
 
+        // Get the default branch
+        let default_branch = repo_stats.default_branch.clone();
+        let default_branch = match default_branch {
+            Some(branch) => branch,
+            None => return Err(anyhow!("No default branch found for package repository")),
+        };
```

Comment:
This could be `let default_branch = default_branch.ok_or_else(|| anyhow!("...."))?;`